apis/python/src/tiledbsoma/_collection.py

```diff
@@ -229,6 +243,24 @@ def add_new_dataframe(
         :meth:`add_new_collection`. The remaining parameters are passed to
         :meth:`DataFrame.create` unchanged.
 
+        Examples:
+            >>> import pandas as pd
+            >>> import pyarrow as pa
+            >>> df = pd.DataFrame(data={"soma_joinid": [0, 1], "col1": [1, 2], "col2": [3, 4]})
+            ... with tiledbsoma.Collection.create("/tmp/collection") as soma_collection:
+            ...     with soma_collection.add_new_dataframe(
+            ...         "a_dataframe", schema=pa.Schema.from_pandas(df)
+            ...     ) as soma_df:
+            ...         soma_df.write(pa.Table.from_pandas(df, preserve_index=False))
```

Comment:
Handling sub-collections no longer needs to be done with a `with` because of lifecycle inheritance:

```python
with tiledbsoma.Collection.create(...) as soma_collection:
  df = soma_collection.add_new_dataframe(...)
  df.write(...)
  # Or, you could make this a one-liner:
  soma_collection.add_new_dataframe(...).write(...)
```

Reply:
thanks - fixed.